From c7dc97c6e4de7255829bbc1146e9944ddc67ed53 Mon Sep 17 00:00:00 2001
From: axtloss <axtlos@disroot.org>
Date: Sun, 19 Jan 2025 02:34:18 +0100
Subject: [PATCH] make: Remove duplicated rules

---
 Makefile | 40 ++++++++++++++++------------------------
 1 file changed, 16 insertions(+), 24 deletions(-)

diff --git a/Makefile b/Makefile
index 4742109..e360a4b 100644
--- a/Makefile
+++ b/Makefile
@@ -12,43 +12,35 @@ MONO_FONT_FILENAMES=IosevkaFixedSS04-Extended.ttf \
 	IosevkaFixedSS04-ExtendedBold.ttf \
 	IosevkaFixedSS04-ExtendedBoldItalic.ttf
 
+
 ADWAITA_FONT_FILENAMES=AdwaitaSans.ttf \
-	AdwaitaSans-Italic.ttf \
-	AdwaitaMono.ttf \
+	AdwaitaSans-Italic.ttf
+
+ADWAITA_MONO_FONT_FILENAMES=AdwaitaMono.ttf
 	AdwaitaMono-Italic.ttf \
 	AdwaitaMono-Bold.ttf \
 	AdwaitaMono-BoldItalic.ttf
 
-all: $(ADWAITA_FONT_FILENAMES)
-
-sans:
-	echo "$(SANS_SHA256) $(SANS_ARCHIVE)" | sha256sum --check --status
-	unzip -o $(SANS_ARCHIVE) $(SANS_FONT_FILENAMES)
+all: $(ADWAITA_FONT_FILENAMES) $(ADWAITA_MONO_FONT_FILENAMES)
 
-AdwaitaSans.ttf: sans
-	pyftfeatfreeze --features "cv05" --replacenames "Inter Variable/Adwaita Sans" InterVariable.ttf $@
 
-AdwaitaSans-Italic.ttf: sans
-	pyftfeatfreeze --features "cv05" --replacenames "Inter Variable/Adwaita Sans" InterVariable-Italic.ttf $@
+$(SANS_FONT_FILENAMES):
+	echo "$(SANS_SHA256) $(SANS_ARCHIVE)" | sha256sum --check --status
+	unzip -o $(SANS_ARCHIVE) $@
 
-mono:
+$(MONO_FONT_FILENAMES):
 	echo "$(MONO_SHA256) $(MONO_ARCHIVE)" | sha256sum --check --status
-	unzip -o $(MONO_ARCHIVE) $(MONO_FONT_FILENAMES)
-
-AdwaitaMono.ttf: mono
-	pyftfeatfreeze --replacenames "Iosevka Fixed SS04/Adwaita Mono" IosevkaFixedSS04-Extended.ttf $@
-
-AdwaitaMono-Italic.ttf: mono
-	pyftfeatfreeze --replacenames "Iosevka Fixed SS04/Adwaita Mono" IosevkaFixedSS04-ExtendedItalic.ttf $@
+	unzip -o $(MONO_ARCHIVE) $@
 
-AdwaitaMono-Bold.ttf: mono
-	pyftfeatfreeze --replacenames "Iosevka Fixed SS04/Adwaita Mono" IosevkaFixedSS04-ExtendedBold.ttf $@
+$(ADWAITA_FONT_FILENAMES): %.ttf: $(SANS_FONT_FILENAMES)
+	pyftfeatfreeze --features "cv05" --replacenames "Inter Variable/Adwaita Sans" $< $@
 
-AdwaitaMono-BoldItalic.ttf: mono
-	pyftfeatfreeze --replacenames "Iosevka Fixed SS04/Adwaita Mono" IosevkaFixedSS04-ExtendedBoldItalic.ttf $@
+$(ADWAITA_MONO_FONT_FILENAMES): %.ttf: $(MONO_FONT_FILENAMES)
+	pyftfeatfreeze --replacenames "Iosevka Fixed SS04/Adwaita Mono" $< $@
 
-install: $(ADWAITA_FONT_FILENAMES)
+install: $(ADWAITA_FONT_FILENAMES) $(ADWAITA_MONO_FONT_FILENAMES)
 	install -Dm655 -t "$(DESTDIR)/$(PREFIX)/share/fonts/Adwaita" $(ADWAITA_FONT_FILENAMES)
+	install -Dm655 -t "$(DESTDIR)/$(PREFIX)/share/fonts/Adwaita" $(ADWAITA_MONO_FONT_FILENAMES)
 
 clean:
 	rm *.ttf
-- 
2.47.0

